app/next-client-app/components/scanreports/MappingStatus.tsx

```diff
@@ -7,58 +7,63 @@ import {
   SelectTrigger,
   SelectValue,
 } from "@/components/ui/select";
-import { statusOptions } from "@/constants/scanReportStatus";
-import { ApiError } from "@/lib/api/error";
+import { MappingStatusOptions } from "@/constants/scanReportStatus";
 import { toast } from "sonner";
 import { cn } from "@/lib/utils";
 import { SelectTriggerProps } from "@radix-ui/react-select";
 
-interface ScanReportStatusProps extends SelectTriggerProps {
+interface MappingStatusProps extends SelectTriggerProps {
   id: string;
-  status: string;
+  mapping_status: Status;
   dataset: string;
   disabled: boolean;
 }
 
-export function ScanReportStatus({
+export function MappingStatus({
   id,
-  status,
+  mapping_status,
   dataset,
   className,
   disabled,
-}: ScanReportStatusProps) {
+}: MappingStatusProps) {
   // Safely extract the color
-  const statusInfo = statusOptions.find((option) => option.value === status);
+  const statusInfo = MappingStatusOptions.find(
+    (option) => option.value === mapping_status.value
+  );
   const textColorClassName = statusInfo?.color ?? "text-black";
 
   const handleChangeStatus = async (newStatus: string) => {
     const response = await updateScanReport(parseInt(id), {
-      status: newStatus,
+      // tried updating data wiht "value" but not successful, needed the id of the new mapping status here
```

Comment:
Is this why you need the Id? If so - we should change the API to allow for values in the mapping status, not need ID.